TEMPLATES.md

```diff
@@ -0,0 +1,103 @@
+# Using `Template`s as Execution Targets With CI Operator
+
+The CI Operator allows for users to configure tests running in a single container
+with the `tests` array. In many cases, however, running a command in a single
+container will not suffice. For more complicated tests, the CI Operator supports
+a `--template` option with which a test author may write black-boxed execution
+targets using OpenShift `Template`s. This document explains the interfaces expected
+for test `Template`s and best practices for how an author should configure their
+`Template`.
+
+The high-level contract for a `Template` execution target is that the `Template`
+will be processed into the test's `Namespace` once prerequisite `Build`s have
+occurred. Artifacts will be gathered from `artifacts` volumes on `Pod`s in
+the `Template`, if any are present. What work occurs in a `Template`'s `Pod`s is
+a black box as far as the CI Operator is concerned.
+
+## Parameters Available to `Template`s
+
+A number of parameters are available to the `Template` and will be provided if
+the `Template` is explicitly configured to consume them. The following parameters
+will be provided to `Template`s:
+
+ - `NAMESPACE`
+   The namespace in which your `Template` is processed. Target this namespace with
+   your objects.
+
+   | Example: | `ci-op-<input-hash>` |
```

Comment:
Because this whole thing is about Templates, it's really important for your examples to be clear when they refer to dynamic data. Right now it's not clear which parts are dynamic and which parts are literal.

Your examples are not (strictly speaking) examples; they're format definitions (I think). If I'm describing the output of `md5sum`, then I might say:

    Output format: <hash> <filename>
    Example: f3a0efce6fa8d4437c0a8392f4622db8 foobar

Your examples seem to combine the two, and mix and match `<thisformat>` and `${thisformat}`. This makes the `IMAGE_FORMAT` definition below confusing to read; it's not clear whether `${component}` is a literal or not.

To improve this, I'd change the "Examples" to actual full-blown examples; don't redact them in any way.